components/server/src/github/github-auth-provider.ts

```diff
@@ -98,26 +98,35 @@ export class GitHubAuthProvider extends GenericAuthProvider {
                 .map((s: string) => s.trim())
             );
 
-            const filterPrimaryEmail = (emails: typeof userEmails) => {
+            const blockPassListEmail = ((emails: typeof userEmails) => {
```

Comment:
I think this should be moved to `UserService.handleNewUser`, cf. https://github.com/gitpod-io/gitpod/blob/main/components/server/src/user/user-service.ts#L140
This way, we do not need to repeat filtering.

Reply:
When we store all verified emails (as suggested below) we can drop this here because it is already in `UserService.handleNewUser`. See:

https://github.com/philschatz/gitpod/blob/88af0c51bf7c29b55e510c24d7c825aaa28b3254/components/server/src/user/user-service.ts#L143-L144

Otherwise, this filter would be necessary to store just the needed email addresses.

Reply:
👍  I believe this is now resolved.